wsd/RequestDetails.cpp

```diff
@@ -283,17 +284,34 @@ Poco::URI RequestDetails::sanitizeURI(const std::string& uri)
 
 std::string RequestDetails::getDocKey(const Poco::URI& uri)
 {
-    // If multiple host-names are used to access us, then
-    // they must be aliases. Permission to access aliased hosts
-    // is checked at the point of accepting incoming connections.
-    // At this point storing the hostname artificially discriminates
-    // between aliases and forces same document (when opened from
-    // alias hosts) to load as separate documents and sharing doesn't
-    // work. Worse, saving overwrites one another.
     std::string docKey;
-    Poco::URI::encode(uri.getPath(), "", docKey);
-    LOG_INF("DocKey from URI [" << uri.toString() << "] => [" << docKey << ']');
+    std::string uriHost = uri.getHost();
+    std::string uriPort = std::to_string(uri.getPort());
+
+    RequestDetails::setUriHostAndPort(uriHost, uriPort);
+
+    Poco::URI::encode(uri.getScheme() + "://" + uriHost + ":" + uriPort + uri.getPath(), "",docKey);
+    LOG_INF("DocKey from URI [" << uri.toString() << "] => [" << docKey << "]");
     return docKey;
 }
 
+void RequestDetails::setUriHostAndPort(std::string& uriHost, std::string& uriPort)
```

Comment:
We could have this function in StorageBase, since it is a static function, accesses static members of StorageBase and really has no relationship to RequestDetails. This will greatly simplify the dependencies and the code. The statics in StorageBase can be private.